### PR TITLE
ASTEROIDS

### PR DESCRIPTION
**Pull request by CDRXavier**
Uploaded proper image, updated HEX, updated info, deleted and recreated **PROPER** folder.
Hmmm.....I deleted one more line than I added.
Binary file diff? Maybe.

### CODE REVIEW

**Comment by CDRXavier** [comment]
Got to change the image size...

**Comment by CDRXavier** [comment]
Image size changed to 128 * 64.
Should have just pushed it...

**Comment by eried** [comment]
Ok, I will check this on sunday and merge it :) thanks for the submit

File: Arcade/ASTEROIDS/game.ini [hunk 1]
```diff
@@ -0,0 +1,5 @@
+title=ASTEROIDS 
+date=2018-06-28T04:41:00Z 
+author=CDR_Xavier 
+description=The unique 2D space shooter. Ammo in the bottomless box.
+license=GNL
```
Comment (Pharap) [inline Arcade/ASTEROIDS/game.ini]: The licence needs to be changed before this is merged anyway.
Currently it says `license=GNL` when it should be `license=GPL-2.0`.

See [here](https://github.com/eried/ArduboyCollection/blob/5fe4c86cfcca04753ed3ac39dce5fc6598e3605c/Arcade/Fatsche/game.ini) and [here](https://github.com/eried/ArduboyCollection/blob/5fe4c86cfcca04753ed3ac39dce5fc6598e3605c/Arcade/Fatsche/game.ini) for comparison.

File: Arcade/ASTEROIDS/game.ini [hunk 2]
```diff
@@ -2,4 +2,4 @@ title=ASTEROIDS
 date=2018-06-28T04:41:00Z 
 author=CDR_Xavier 
 description=The unique 2D space shooter. Ammo in the bottomless box.
-license=GNL
+license=GNL 2.0
```
Comment (Pharap) [inline Arcade/ASTEROIDS/game.ini]: Still not quite right.
Now it says "GNL 2.0", it needs to be "GPL-2.0".

File: Arcade/ASTEROIDS/game.ini [hunk 3]
```diff
@@ -0,0 +1,5 @@
+title=ASTEROIDS 
+date=2018-06-28T04:41:00Z 
+author=CDR_Xavier 
+description=The unique 2D space shooter. Ammo in the bottomless box.
+license=GNL-2.0
```
Comment (Pharap) [inline Arcade/ASTEROIDS/game.ini]: Still not right.
Now it says `GNL-2.0`, but it needs to say `GPL-2.0`.
It is the GNU **G**eneral **P**ublic **L**icence version **2.0**, hence `GPL-2.0`.

File: Arcade/SNAKE/game.ini [hunk 4]
```diff
@@ -0,0 +1,5 @@
+title=ASTEROIDS
+date=2018-07-07T07:07:07Z
+author=CDR_Xavier
+description=The snake as it appears on your Nokia screen. Not eating apples.
+license=GNL-2.0
```
Comment (Pharap) [inline Arcade/SNAKE/game.ini]: Likewise, this is also not not right for the same reason.
`GPL-2.0`, not `GNL-2.0`.

**Comment by Pharap** [comment]
The licences are indeed correct this time.

**Comment by Pharap** [comment]
@eried, when it comes to merging this, it might be best to [squash the commits](https://blog.github.com/2016-04-01-squash-your-commits/) so it ends up being 1 big commit instead of 25 small commits.

**Comment by CDRXavier** [comment]
How to do that?

**Comment by CDRXavier** [comment]
"To merge pull requests, you must have write permissions in the repository."
Ok. I don't have write access to this...

**Comment by Pharap** [comment]
@CDRXavier only @eried can merge it.

And he can't merge it if you've closed it.
You need to reopen it and leave it until @eried has time to merge it.

**Comment by CDRXavier** [comment]
lol...
I was thinking that it was my problem and worrying about crap...

**Comment by Pharap** [comment]
Where I live it's 15:48, and where eried lives it's 16:48.
There's still plenty of Sunday left.

The reason only eried can merge is because it's his repo and he hasn't given anyone else write permission.
There's no rush anyway.

**Comment by eried** [comment]
Yeah, I'm back but I haven't checked your PR yet. I will merge in few hours

**Comment by eried** [comment]
I tried to add your games @CDRXavier but there are too many mistakes. The idea about using git is that adding games should be easy, not days of discussion and a big fixing task to me or anyone.

Please:
1) READ the instructions on the main page https://github.com/eried/ArduboyCollection print them and follow them one by one. Use a pen to draw a tick after you do a step.
2) Do few commits, with some significant changes 
3) Please **choose a different folder**, `Snake` is already used by an `FManga` game (again, follow step 1)
4) No extra spaces in the `game.ini` ending (follow step 1, use a text editor that show you spaces if necessary)

**Comment by eried** [comment]
I tried to add your games @CDRXavier but there are too many mistakes. The idea about using git is that adding games should be easy, not days of discussion and a big fixing task to me or anyone.

Please:
1) READ the instructions on the main page https://github.com/eried/ArduboyCollection print them and follow them one by one. Use a pen to draw a tick after you completed each step.
2) Do few commits, with some significant changes 
3) Please **choose a different folder**, `Snake` is already used by an `FManga` game (again, follow step 1)
4) No extra spaces in the `game.ini` ending (follow step 1, use a text editor that show you spaces if necessary)

**Comment by Pharap** [comment]
Perhaps I ought to handle adding these games, since I've had more experience with it?
(Not to say I don't make mistakes, but when I do they're usually small typos.)

I'm not sure how to handle the name clash with 'Snake' though.
Would using allcaps definitely confuse the system?

It's kind of a shame that the system groups by category instead of by author.
Grouping by author would avoid the name clash issue.

**Comment by eried** [comment]
How do you prevent the creation of invalid Categories? I think git is already a bit out of the scope of acting as a repo (using a game.ini is already a stretch on my original idea of KISS)

About adding the games yourself, I am not sure. It will prevent @CDRXavier from learning. I am not sure how much simple I can make the process/instructions.

**Comment by CDRXavier** [comment]
Well I saw the snake game (the FManga game) and the folder is called Snake, but my folder is SNAKE (capitalization, but if you count them as similar...then sure, should be more creative when coming up with folder names...)
Few commits with significant changes. Like, how?
There are plenty of moving around involved, and updating the existing code.
That is why it involve 13 files (although 25 commits may seem to be too much for change in 13 files?)
"No extra Space"--Well, for that I can't tell just by looking at GitHub... I know it is annoying, but I can't see them...Yes, there are text editors that show "unprintable characters" and I have one, but I am not feeling to do such "fancy" things. Will use directional keys to test them later on.
About the directions, I am pretty sure I didn't miss any or were doing anything wrong. (I could be blind sometimes, though.)
As of category, I think it is a smaller matter. (But in this case, examples are already there--"**ATEROIDS**", Snake--and they guided me quite fine.) Probably will talk to @eried if any issue over category emerge. :)
Cheers.

**Comment by CDRXavier** [comment]
Would using allCaps definitely confuse the system?
I don't know, but from what I have seen they ARE case-sensitive.
I could change the names of folders, if one wish.
But yeah, the ASTEROID's success was based on the fact that ~~there are no previous asteroid game.~~ akkera102 used "Tiny Asteroid" as name.
Call that Big Snake?

**Comment by Pharap** [comment]
> How do you prevent the creation of invalid Categories?

To answer that, you must first be able to define which categories are valid and which are invalid.

You could use a mechanism similar to the current licence system:
The author/owner of the repo dictates a whitelist of categories,
and new categories must be requested and approved to be recognised.

>  It will prevent @CDRXavier from learning.

Hrm, that's a fair point.

> I am not sure how much simple I can make the process/instructions.

I think the issue is more that @CDRXavier isn't used to using Github yet.
He's only really used it for hosting games up until recently.
All the fork/branch/PR stuff might be second-nature to you and I, but it's easy to forget how confusing Github is when you don't understand how it works.
(Parts of it are still confusing even when you do know how it works.)

> Call that Big Snake

What about CDRSnake or XSnake?

**Comment by eried** [comment]
> Would using allCaps definitely confuse the system?
> I don't know, but from what I have seen they ARE case-sensitive.
> I could change the names of folders, if one wish.
> But yeah, the ASTEROID's success was based on the fact that there are no previous asteroid game. > > > akkera102 used "Tiny Asteroid" as name.

You can use the same name inside the `game.ini` if you wish, but the folder has to be different. Capitalization will create issues in Windows, for anyone trying to fork the repo in the future. Sorry but I forgot to add that limitation in the `readme`

> Few commits with significant changes. Like, how?

How are you creating the files? via web in github itself? It should be 1-2 commits per game, you can add several related games in 1 commit

> "No extra Space"--Well, for that I can't tell just by looking at GitHub... 

True. Also it was not explicit as a limitation in the `readme `again

**Comment by eried** [comment]
OK, for now @CDRXavier I have duplicated your PR into a new branch https://github.com/eried/ArduboyCollection/pull/55 and your games are added now, check them out in http://arduboy.ried.cl

No need to do anything else.

**Pull request merged**